### PR TITLE
GitCommit.raw_log only drop last msg line if blank

### PR DESCRIPTION
**Pull request by scottnonnenberg**
Fixes the message-dropping behavior I saw, potentially the same thing as #38.

### CODE REVIEW

**Comment by jankeromnes** [comment]
Hi @scottnonnenberg, thanks a lot for fixing this! Do you think we could also add a test that ensures one-line commit messages aren't being dropped in the future?

File: lib/git/internal/git_commit.js [hunk 1]
```diff
@@ -65,9 +65,15 @@ GitCommit.from_raw = function(raw_object, repository) {
...
+  var lines = this.message.split("\n");
+  var length = lines.length;
+
+  // drop last line if it's blank
+  if (length > 0 && lines[length - 1].length === 0) {
+    length -= 1;
+  }
+
+  for(var i = 0; i < length; i++) {
     output = output + '    ' + lines[i] + '\n';
```
Comment (jankeromnes) [inline lib/git/internal/git_commit.js]: Nit: Please use `output += ` here.

File: lib/git/internal/git_commit.js [hunk 2]
```diff
@@ -65,9 +65,15 @@ GitCommit.from_raw = function(raw_object, repository) {
...
   output = output + this.headers + "\n\n";
-  var lines = this.message.split("\n");    
-  // Remove the last line which will be empty
-  for(var i = 0; i < (lines.length > 1 ? lines.length - 1 : lines.length); i++) {
+  var lines = this.message.split("\n");
+  var length = lines.length;
+
+  // drop last line if it's blank
+  if (length > 0 && lines[length - 1].length === 0) {
+    length -= 1;
```
Comment (jankeromnes) [inline lib/git/internal/git_commit.js]: Nit: Please use `length--` here.

**Comment by scottnonnenberg** [comment]
Unit tests added. Seems that the couple full repos under /test all have the expected trailing newline, so I couldn't use any existing fixture data.

**Comment by jankeromnes** [comment]
Awesome, thanks!

**Pull request merged**